docs/manual/mustcall-checker.tex

```diff
@@ -0,0 +1,164 @@
+\htmlhr
+\chapterAndLabel{Must Call Checker}{mustcall-checker}
+
+The MustCall Checker tracks the methods that an object should call before it is deallocated
+and produces a conservative overapproximation of the set of methods that might
+actually need to be called on an object.
+It is intended to be run as a subchecker of another checker: on its own, it does not
+enforce any rules other than subtyping.
+
+For example, an object whose type is \<java.io.OutputStream> might
+have an obligation to call the \<close()> method, to release an underlying file resource. Or,
+such an \<OutputStream> might not have such an obligation, if the underlying resource is
+a byte array. Either of these obligations can be represented by the static type
+\<@MustCall(\{"close"\}) OutputStream>, which can be read as ``an OutputStream that might need
+to call close before it is deallocated''.
+
+\sectionAndLabel{MustCall annotations}{must-call-annotations}
+
+The MustCall Checker supports these qualifiers and declaration annotations:
+
+\begin{description}
+
+\item[\refqualclasswithparams{checker/mustcall/qual}{MustCall}{String[] value}]
+  indicates that the qualifier's arguments are the methods that the annotated type
+  should call. The type \<@MustCall(\{"a"\}) Object obj> means
+  that before \<obj> is deallocated, \<obj.a()> should be called.
+  The default type qualifier in unannotated code is this type with an empty argument.
+
+\item[\refqualclass{checker/mustcall/qual}{MustCallUnknown}]
+  is used internally by the type system but should never be written by a
+  programmer.  It theoretically represents a value with an unknowable or infinite set
+  of must-call methods.
+
+\item[\refqualclasswithparams{checker/mustcall/qual}{InheritableMustCall}{String[] value}]
+  is an alias of \<@MustCall> that can only be written on a class declaration. It adds an \<@MustCall>
+  annotation with the same arguments to
+  the class declaration on which it is written, and also to all subclasses, freeing the user of the need
+  to write the \<@MustCall> annotation on every subclass.
+
+\item[\refqualclass{checker/mustcall/qual}{PolyMustCall}]
+  is the polymorphic annotation for the MustCall type system.
+
+\item[\refqualclass{checker/mustcall/qual}{MustCallAlias}]
+  is a special polymorphic annotation that indicates a resource-aliasing relationship between
```

Comment:
"resource-aliasing" is a term we define in the paper, but (like I mentioned above wrt examples) it doesn't make much sense in the context of the Must Call Checker, which does treat it identically to `@PolyMustCall`. I'm okay with the term being used but not defined with the intention of defining it when the consistency checker is upstreamed, but I'd also be okay with removing all references to it and just saying that `@MustCallAlias` is an alias for `@PolyMustCall` and not explaining anything. Alternatively, we could just remove the references to `@MustCallAlias` in the manual altogether, and plan to document it when the consistency checker is upstreamed, since someone who is only using the Must Call Checker doesn't need it; I'm okay with that too.